### PR TITLE
Fix #1589 : Naming consistency for DataProviderIds in doma…

### LINKED ISSUE #1589: Naming consistency for DataProviderIds in domain layer.
In `domain` layer all data provider ids should follow consistent pattern.
We should follow below mentioned style:

 - Name of constant & value should be the same (e.g. `QUESTION_DATA_PROVIDER_ID` and `"question_data_provider_id"`)
 - Constant name should be UPPER_SNAKE_CASE and value should be lower_snake_case
 - Use ID format: `<function_name>_<optional_context>_PROVIDER_ID` where the function_name is the exact function the provider used (expected to be just 1) and optional_context is used to disambiguate either when the ID is used in different functions or if the same function is overloaded

### PR DESCRIPTION
**Pull request by The-Fuse**
## Explanation
Fix #1589 
Naming consistency for DataProviderIds in domain-layer
In domain layer all data provider ids should follow consistent pattern.
We should follow below mentioned style:

Name of constant & value should be the same (e.g. QUESTION_DATA_PROVIDER_ID and "question_data_provider_id")
Constant name should be UPPER_SNAKE_CASE and value should be lower_snake_case
Use ID format: <function_name>_<optional_context>_PROVIDER_ID where the function_name is the exact function the provider used (expected to be just 1) and optional_context is used to disambiguate either when the ID is used in different functions or if the same function is overloaded

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by The-Fuse** [comment]
@rt4914  Please review this.

File: domain/src/main/java/org/oppia/domain/exploration/ExplorationDataController.kt [hunk 1]
```diff
@@ -9,7 +9,7 @@ import org.oppia.util.data.DataProviders
 import org.oppia.util.system.OppiaClock
 import javax.inject.Inject
 
-private const val EXPLORATION_DATA_PROVIDER_ID = "ExplorationDataProvider"
+private const val EXPLORATION_DATA_PROVIDER_ID = "exploration_data_provider_id"
```
Comment (rt4914) [inline domain/src/main/java/org/oppia/domain/exploration/ExplorationDataController.kt:12]: This does not following the convention mentioned in the issue.
`<function_name>_<optional_context>_PROVIDER_ID`
means that the rename should be 
`GET_EXPLORATION_BY_ID_PROVIDER_ID ="get_exploration_by_id_provider_id"`

**Review by rt4914 — CHANGES_REQUESTED** [review]
Thanks for submitting PR.
The convention for the renaming is `<function_name>_<optional_context>_PROVIDER_ID` which means the id should start with a function name where it is getting used.

Also, you have missed some files like `StoryProgressController`, so make sure you check the entire domain module.


Also,
Your PR title should be `Fix #1589: Nam.....`, meaning you should add `:` after issue number.

Also, make sure that your branch name is of lower case separated by `-`, meaning `naming-consistency-for-data-provider-ids`

**Comment by anandwana001** [comment]
@The-Fuse any update on this?

**Comment by The-Fuse** [comment]
Yes , working on it .

**Comment by anandwana001** [comment]
@The-Fuse Let me know if you need any assistance on this or if you are finding any difficulty in the codebase.

**Pull request closed without merging**